Signal/src/ViewControllers/ContactsViewHelper.m

```diff
@@ -26,19 +26,26 @@ @interface ContactsViewHelper ()
 
 @property (nonatomic) NSArray<NSString *> *blockedPhoneNumbers;
 
+@property (nonatomic) BOOL suppressDelegateEvents;
+
 @end
 
 #pragma mark -
 
 @implementation ContactsViewHelper
 
-- (instancetype)init
+- (instancetype)initWithDelegate:(id<ContactsViewHelperDelegate>)delegate
 {
     self = [super init];
     if (!self) {
         return self;
     }
 
+    OWSAssert(delegate);
+    _delegate = delegate;
+
+    self.suppressDelegateEvents = YES;
```

Comment:
Why do we want the "shouldHideLocalNumber" method, it's not obvious that we're using it during init?

And there are only two delegate methods at the moment, so when you say we don't want to fire delegate events except using the shouldHideLocalNumber, you mean we don't want to fire the other one, which is just contactsViewHelperDidUpdateContacts, right?

I imagine this makes sense to you after digging a bit, but at first glance I don't know what behavior you're trying to fix. Can you leave a code comment that outlines the rationale?